lib/src/impl_js/impl_js.digest.dart

```diff
@@ -35,14 +35,9 @@ class _Hash implements Hash {
   }
 }
 
-const Hash sha1 = _Hash('SHA-1');
-const Hash sha256 = _Hash('SHA-256');
-const Hash sha384 = _Hash('SHA-384');
-const Hash sha512 = _Hash('SHA-512');
-
 /// Get the algorithm from [hash] or throw an [ArgumentError].
-String _getHashAlgorithm(Hash hash) {
-  if (hash is _Hash) {
+String _getHashAlgorithm(HashImpl hash) {
+  if (hash is _HashImpl) {
     return hash._algorithm;
   }
   throw ArgumentError.value(
```

Comment:
Let's also make this an `AssertionError`, because it shouldn't be possible, right?

Reply:
Added `Assertion Error` with the same message